buildSrc/src/main/java/com/scalar/db/JdkConfigurationPlugin.java

```diff
@@ -0,0 +1,152 @@
+package com.scalar.db;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.api.tasks.testing.Test;
+import org.gradle.jvm.toolchain.JavaLanguageVersion;
+import org.gradle.jvm.toolchain.JavaToolchainService;
+import org.gradle.jvm.toolchain.JvmVendorSpec;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Plugin configuring all tasks and integration test tasks to use a given JDK defined by the
+ * following Gradle properties:
+ *
+ * <ul>
+ *   <li>`javaVersion` : configure all Java related tasks to use the given JDK version. The default
+ *       java version 8.
+ *   <li>`javaVendor` : configure all Java related tasks to use the given JDK vendor. The default
+ *       java vendor is Adoptium (also known as Temurin).
+ *   <li>`integrationTestJavaRuntimeVersion` : configure all test tasks name starting with
+ *       "integrationTest" to run with the given JDK version. The default Java runtime version is 8.
+ *   <li>`integrationTestJavaRuntimeVendor` : configure all test tasks name starting with
+ *       "integrationTest" to run with the given JDK vendor. The default Java runtime vendor is
+ *       Adoptium (also known as Temurin).
+ * </ul>
+ *
+ * <p>Usage example using the CLI:
+ *
+ * <p>1. To use JDK 11 (amazon) for all Java tasks including integration tests
+ *
+ * <pre><code>
+ *   gradle integrationTestJdbc -PjavaVersion=11 -PjavaVendor=amazon
+ * </code></pre>
+ *
+ * 2. To use JDK 11 (amazon) for all Java tasks while having integration test use JDK 17 (microsoft)
+ *
+ * <pre><code>
+ *   gradle integrationTestJdbc -PjavaVersion=11 -PjavaVendor=amazon -PintegrationTestJavaRuntimeVersion=17 -PintegrationTestJavaRuntimeVendor=microsoft
+ * </code></pre>
+ */
+public class JdkConfigurationPlugin implements Plugin<Project> {
```

Comment:
I added a way to configure the SDK used for all Java tasks (compilation, unit tests, integration tests, etc...). 
```
gradle integrationTestJdbc -PjavaVersion=11 -PjavaVendor=amazon
```
This differs from [what we decided earlier](https://github.com/scalar-labs/scalardb/pull/1749#discussion_r1618065646) because if we could only configure the compiler JDK, we could run into an issue where the test tasks fail to run because it uses a JDK version anterior to the compiler JDK version. So, for simplicity, we configure all Java tasks to use a common JDK, but we can still set a specific JDK to run the integration test with:
```
gradle integrationTestJdbc -PintegrationTestJavaRuntimeVersion=17 -PintegrationTestJavaRuntimeVendor=microsoft
```